presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -130,6 +133,7 @@
     private final ReentrantReadWriteLock taskExecutionLock = new ReentrantReadWriteLock();
 
     private HiveSplitSource hiveSplitSource;
+    private AtomicReference<Optional<PathFilter>> pathFilterReference = new AtomicReference<>();
```

Comment:
I need this AtomicReference since loadingPartitions is a multi threaded implementation. Ideally  the first thread that processes the partitions would set the PathFilter and rest of the threads can simply use it. I kept it as AtomicReference since we wanted the FileFormatSpecificPathFilterFactory.createPathFilter(inputFormatName) to take in a String inputFormatName as a parameter. To identify this inputformat name inside the  BackgroundSplitLoader constructor may be a bit hacky. Instead if the createPathFilter(inputFormatName) can be changed to getPathFilter() then we wouldnt need the AtomicReference (We can directly assign the PathFilter in the BackgroundSplitLoader constructor). Please let me know if you prefer one way or other.

Reply:
The `BackgroundHiveSplitLoader` loads splits for multiple partitions. Partitions are allowed to have distinct storage format. Why not simply recreated the filter for every partition? Is it not a lightweight operation enough?